File: CTranslator/src/Token.cs
using System;

namespace CTranslator
{
    public enum TokenType
    {
        // Ключевые слова
        KW_INT, KW_IF, KW_ELSE, KW_SCANF, KW_PRINTF, KW_RETURN,
        // Идентификатор и литералы
        IDENTIFIER, NUMBER, STRING_LIT,
        // Арифметические операторы
        OP_PLUS, OP_MINUS, OP_STAR, OP_SLASH,
        // Операторы сравнения и присваивания
        OP_ASSIGN, OP_EQ, OP_NEQ,
        OP_LT, OP_GT, OP_LE, OP_GE,
        // Разделители
        LPAREN, RPAREN, LBRACE, RBRACE,
        SEMICOLON, COMMA, AMP,
        // Служебные
        MARKER, EOF
    }

    public class Token
    {
        public TokenType Type   { get; }
        public string    Value  { get; }
        public int       Line   { get; }
        public int       Column { get; }

        public Token(TokenType type, string value, int line, int column)
        {
            Type = type; Value = value; Line = line; Column = column;
        }

        public override string ToString() =>
            $"{Type,-15} {Value,-15} L{Line} C{Column}";
    }
}

File: src/token.c
/*
 * token.c — реализация вспомогательных функций для токенов
 */
#include "token.h"

const char *token_type_name(TokenType t)
{
    switch (t) {
        case KW_INT:      return "KW_INT";
        case KW_IF:       return "KW_IF";
        case KW_ELSE:     return "KW_ELSE";
        case KW_SCANF:    return "KW_SCANF";
        case KW_PRINTF:   return "KW_PRINTF";
        case KW_RETURN:   return "KW_RETURN";
        case TK_IDENT:    return "IDENTIFIER";
        case TK_NUMBER:   return "NUMBER";
        case TK_STRING:   return "STRING";
        case OP_PLUS:     return "OP_PLUS";
        case OP_MINUS:    return "OP_MINUS";
        case OP_STAR:     return "OP_STAR";
        case OP_SLASH:    return "OP_SLASH";
        case OP_EQ:       return "OP_EQ";
        case OP_NEQ:      return "OP_NEQ";
        case OP_LT:       return "OP_LT";
        case OP_GT:       return "OP_GT";
        case OP_LE:       return "OP_LE";
        case OP_GE:       return "OP_GE";
        case OP_ASSIGN:   return "OP_ASSIGN";
        case TK_LPAREN:   return "LPAREN";
        case TK_RPAREN:   return "RPAREN";
        case TK_LBRACE:   return "LBRACE";
        case TK_RBRACE:   return "RBRACE";
        case TK_SEMICOLON:return "SEMICOLON";
        case TK_COMMA:    return "COMMA";
        case TK_AMP:      return "AMP";
        case TK_MARKER:   return "MARKER";
        case TK_EOF:      return "EOF";
        default:          return "UNKNOWN";
    }
}
